File: DSA With SUPREME 2.0 Batch/Array/PrintPair.cpp
//Example : I/P -> [1,2,3] 
// O/P -> [1,1],[1,2],[1,3] [2,1],[2,2],[2,3]  [3,1],[3,2],[3,3]
#include<iostream>
using namespace std;

void print_AlltwoPairs(int arr[], int size){
    for(int i=0 ; i<size ; i++){
        for(int j=0 ; j<size ; j++){
            cout<<arr[i]<<","<<arr[j]<<" ";
        }
        cout<<endl;
    }
}

void print_AllthreePairs(int arr[], int size){
    for(int i=0 ; i<size ; i++){
        for(int j=0 ; j<size ; j++){
            for(int k=0 ; k<size ; k++){
            cout<<arr[i]<<","<<arr[j]<<","<<arr[k]<<" ";
            }
        }
        cout<<endl;
    }
}

int main(){
    int arr[] ={1,2,3};
    print_AlltwoPairs(arr,3);
    // print_AllthreePairs(arr,3);
    return 0;
}

File: DSA With SUPREME 2.0 Batch/01Array/PrintPair.java
/*
* Example : I/P -> [1,2,3] 
O/P -> [1,1],[1,2],[1,3] [2,1],[2,2],[2,3] [3,1],[3,2],[3,3]
*/

class PrintPair{

    public static void printPair(int[] arr){

        if(arr == null || arr.length == 0){
            throw new IllegalArgumentException("Array must not be null");
        }

        for(int i=0; i<arr.length; i++){
            for(int j=0; j<arr.length; j++){
                // Print each pair arr[i], arr[j]
                System.out.println(arr[i]+","+arr[j]);
            }
        }
    }

    public static void main(String[] args)
    {
        int[] arr = {1,2,3};
        printPair(arr);
    }
}